mne/source_space.py

```diff
@@ -3197,3 +3199,90 @@ def _get_src_nn(s, use_cps=True, vertices=None):
     else:
         nn = s['nn'][vertices, :]
     return nn
+
+
+@verbose
+def vertex_depths(src, info=None, picks=None, trans=None, mode=None,
+                  verbose=None):
+    """Compute source depths of vertices.
+
+    Parameters
+    ----------
+    src : instance of SourceSpaces
+        The object to select vertices from.
+    info : instance of Info | None
+        If not None, the depth of each vertex will be computed as its minimum
+        distance to sensors specified by 'info' and 'picks'. If none, the depth
+        will be computed as the distance to the mean location (centre of
+        gravity) of all vertices in source space.
```

Comment:
I'd rather see the (distance to sensors / distance to centre of gravity) be implemented in two different functions. Switching between these modes based on whether info is `None` is too magical for my tastes. You can also see that now the function body is one giant if statement, which hints that it should really be two functions.

Reply:
I agree, also because distance to COG and distance to sensors are anticorrelated, which may be confusing. 
What about
compute_distance_from_sensors()
compute_distance_from_centre()
?